src/impl/conv_im2col.jl

```diff
@@ -44,16 +45,20 @@ function conv_im2col!(
     N = channels_out(cdims)
     K = prod(kernel_size(cdims))*channels_in(cdims)
 
-    @threads for batch_idx in 1:size(x,5)
-        # col_slice is a thread-local workspace
-        col_slice = view(col, :, :, threadid())
-
-        im2col!(col_slice, view(x, :, :, :, :, batch_idx), cdims)
-        GC.@preserve col_slice w y begin
-            col_ptr = pointer(col_slice)
-            w_ptr = pointer(w)
-            y_ptr = pointer(y, (batch_idx - 1)*M*N + 1)
-            gemm!(Val(false), Val(false), M, N, K, alpha, col_ptr, w_ptr, beta, y_ptr)
+    parts = collect(Iterators.partition(axes(x, 5), ceil(Int, size(x, 5) / ntasks)))
```

Comment:
How much effort would it take to avoid these `collect`s?

Reply:
I don't think it's worth putting effort into avoiding allocations here, this is just partitioning indexes

Reply:
My question was more why line 50 below couldn't be:
```julia
@sync for (task_n, part) in enumerate(parts)
```

Reply:
I had tried `pairs` which didn't work with Iterators, but `enumerate` is good. Thanks. Updated